install/kubernetes/cilium/templates/clustermesh-apiserver/deployment.yaml

```diff
@@ -128,42 +126,33 @@ spec:
             configMapKeyRef:
               key: identity-allocation-mode
               name: cilium-config
-{{- with .Values.clustermesh.apiserver.resources }}
-        resources: {{- toYaml . | nindent 10 }}
-{{- end }}
+        {{- with .Values.clustermesh.apiserver.resources }}
+        resources:
+          {{- toYaml . | nindent 10 }}
+        {{- end }}
         volumeMounts:
-        - mountPath: /var/lib/cilium/etcd-secrets
-          name: etcd-admin-client
+        - name: etcd-admin-client
+          mountPath: /var/lib/cilium/etcd-secrets
           readOnly: true
       volumes:
       - name: etcd-server-secrets
-        projected:
+        secret:
+          secretName: clustermesh-apiserver-server-cert
           defaultMode: 0420
-          sources:
-          - secret:
-              name: clustermesh-apiserver-ca-cert
-              items:
-              - key: ca.crt
-                path: ca.crt
-          - secret:
-              name: clustermesh-apiserver-server-cert
```

Comment:
@gandro I think you have a little mistake here: CA private key is `ca.key`, not `tls.key`. `tls.key` is private key of certificate `tls.crt`

Reply:
Two approaches are the same, it contains 3 files `ca.crt`, `tls.crt`, `tls.key`. and no CA private key (`ca.key`).
```yaml
      - name: etcd-server-client
        projected:
          defaultMode: 0420
          sources:
          - secret:
              name: clustermesh-apiserver-ca-cert
              items:
              - key: ca.crt
                path: ca.crt
          - secret:
              name: clustermesh-apiserver-server-cert
```
mean the volume contains:
* `ca.crt` from `clustermesh-apiserver-ca-cert`, `ca.key` is not mount because it's not in `items` list
* mount all files from `clustermesh-apiserver-server-cert` except `ca.cert` because it's overridden by `ca.crt` from `clustermesh-apiserver-ca-cert`

And because `ca.crt` from `clustermesh-apiserver-ca-cert` and `clustermesh-apiserver-server-cert` are the same. So we can eliminate `clustermesh-apiserver-ca-cert` and simplify as bellow:
```yaml
      - name: etcd-server-secrets
        secret:
          secretName: clustermesh-apiserver-server-cert
          defaultMode: 0420
```

Reply:
Ohh, you are 100% correct. Good catch - thank you!